### PR TITLE
PERF: readability container size empty

### PR DESCRIPTION
**Pull request by hjmjohnson**
The emptiness of a container should be checked using the empty() method
instead of the size() method. It is not guaranteed that size() is a
constant-time function, and it is generally more efficient and also
shows clearer intent to use empty(). Furthermore some containers may
implement the empty() method but not implement the size() method. Using
empty() whenever possible makes it easier to switch to another container
in the future.

SRCDIR=/Users/johnsonhj/Dashboard/src/ITK #My local SRC
BLDDIR=/Users/johnsonhj/Dashboard/src/ITK-clang #My local BLD

cd /Users/johnsonhj/Dashboard/src/ITK-clang
run-clang-tidy.py -extra-arg=-D__clang__ -checks=-*,readability-container-size-empty  -header-filter=.* -fix

### CODE REVIEW

File: Modules/IO/GDCM/src/itkGDCMImageIO.cxx [hunk 1]
```diff
@@ -720,7 +720,7 @@ bool GDCMImageIO::CanWriteFile(const char *name)
 {
   std::string filename = name;
 
-  if (  filename == "" )
+  if (  filename.empty() )
```
Comment (dzenanz) [inline Modules/IO/GDCM/src/itkGDCMImageIO.cxx]: Do you intend to contribute 3rd party (GDCM etc) changes upstream?

**Comment by dzenanz** [comment]
@thewtex I don't see build errors from this PR on the [dashboard](https://open.cdash.org/index.php?project=Insight). Was there some recent change, or is something wrong?

**Comment by thewtex** [comment]
> @thewtex I don't see build errors from this PR on the dashboard. Was there some recent change, or is something wrong?

I am not following - are errors expected?

**Comment by hjmjohnson** [comment]
@thewtex @dzenanz There was some CI build anomaly where all builds failed.  I did a simple rebase on upstream/master to force used which triggered successful CI builds.

**Pull request merged**